client/state/document/reducer.js

```diff
@@ -0,0 +1,51 @@
+/**
+ * External dependencies
+ */
+import { combineReducers } from 'redux';
+
+/**
+ * Internal dependencies
+ */
+import {
+	SET_DOCUMENT_DESCRIPTION,
+	SET_DOCUMENT_TITLE,
+} from 'state/action-types';
+
+const INITIAL_STATE = {
+	title: '',
+	description: '',
+};
+
+export function title( state = '', action ) {
+	switch ( action.type ) {
+		case SET_DOCUMENT_TITLE:
+			return action.title;
+	}
+
+	return state;
+}
+
+export function description( state = '', action ) {
```

Comment:
>And while we are at it, I think we should also add the `link`, `script` and `base` tags.

I'm not sure we need to go so far so early, except `<link>`, since that's currently needed for canonical URL.